src/sidebar/components/test/OpenDashboardMenuItem-test.js

```diff
@@ -0,0 +1,72 @@
+import { mount } from 'enzyme';
+import sinon from 'sinon';
+
+import OpenDashboardMenuItem from '../OpenDashboardMenuItem';
+
+describe('OpenDashboardMenuItem', () => {
+  let fakeDashboard;
+
+  beforeEach(() => {
+    fakeDashboard = {
+      getAuthToken: sinon.stub().resolves('auth_token'),
+      open: sinon.stub(),
+    };
+  });
+
+  function createComponent({ isMenuOpen = false } = {}) {
+    return mount(
+      <OpenDashboardMenuItem
+        isMenuOpen={isMenuOpen}
+        dashboard={fakeDashboard}
+      />,
+    );
+  }
+
+  context('when menu is closed', () => {
+    it('does not try to load auth token', () => {
+      createComponent();
+      assert.notCalled(fakeDashboard.getAuthToken);
+    });
+
+    it('has disabled menu item', () => {
+      const wrapper = createComponent();
+      assert.isTrue(wrapper.find('MenuItem').prop('isDisabled'));
+    });
+
+    it('does not open dashboard when item is clicked', () => {
+      const wrapper = createComponent();
+
+      wrapper.find('MenuItem').props().onClick();
+      assert.notCalled(fakeDashboard.open);
+    });
+  });
+
+  context('when menu is open', () => {
+    async function createOpenComponent() {
+      const wrapper = createComponent({ isMenuOpen: true });
+
+      // Wait for auth token call to settle
+      await fakeDashboard.getAuthToken.lastCall.returnValue;
+      wrapper.update();
```

Comment:
Tests are usually more robust to changes in details of async behavior if they wait for the visible effects of an async action, such as waiting for the menu item to become enabled.

Reply:
True!